cmd/mt-kafka-mdm-report-out-of-order/flags.go

```diff
@@ -0,0 +1,135 @@
+		path = flags.Config
+	}
+	config, err := globalconf.NewWithOptions(&globalconf.Options{
+		Filename:  path,
+		EnvPrefix: "MT_",
+	})
+	if err != nil {
+		log.Fatalf("error with configuration file: %s", err.Error())
+		os.Exit(1)
+	}
+	_ = cassandra.ConfigSetup()
+	inKafkaMdm.ConfigSetup()
+	config.Parse()
+
+	flags.RunDuration, err = time.ParseDuration(flags.runDurationStr)
+	if err != nil {
+		log.Fatalf("failed to parse run duration %s: %s", flags.runDurationStr, err.Error)
+		os.Exit(1)
+	}
+
+	if flags.GroupByName == false && flags.GroupByTag == "" {
+		log.Fatalf("must specify one of -group-by-name or -group-by-tag")
+		os.Exit(1)
+	}
+}
+
+func (flags *Flags) Usage() {
+	fmt.Fprintln(os.Stderr, "mt-kafka-mdm-report-out-of-order")
+	fmt.Fprintln(os.Stderr)
+	fmt.Fprintln(os.Stderr, "Inspects what's flowing through kafka (in mdm format) and reports out of order data grouped by metric name or tag, taking into account the reorder buffer)")
+	fmt.Fprintln(os.Stderr)
+	fmt.Fprintln(os.Stderr, "# Mechanism")
+	fmt.Fprintln(os.Stderr, "* it sniffs points being added on a per-series (metric Id) level")
+	fmt.Fprintln(os.Stderr, "* for every series, tracks the last 'correct' point.  E.g. a point that was able to be added to the series because its timestamp is higher than any previous timestamp")
+	fmt.Fprintln(os.Stderr, "* if for any series, a point comes in with a timestamp equal or lower than the last point correct point - which metrictank would not add unless it falls within the reorder buffer - it triggers an event for this out-of-order point")
+	fmt.Fprintln(os.Stderr, "* the reorder buffer is described by the metric interval and the window size")
+	fmt.Fprintln(os.Stderr, "Usage:")
+	fmt.Fprintln(os.Stderr, "  mt-kafka-mdm-report-out-of-order [flags]")
+	fmt.Fprintln(os.Stderr)
+	fmt.Fprintln(os.Stderr, "out-of-order metrics are printed in the following format based on grouping")
+	fmt.Fprintln(os.Stderr)
+	fmt.Fprintln(os.Stderr, "# Grouped by Names")
+	fmt.Fprintln(os.Stderr, "name=<name1> count=<count1>")
+	fmt.Fprintln(os.Stderr, "name=<name1> count=<count1>")
+	fmt.Fprintln(os.Stderr, "...")
+	fmt.Fprintln(os.Stderr, "# Grouped by Tags")
+	fmt.Fprintln(os.Stderr, "tag=<tag1> count=<count1>")
+	fmt.Fprintln(os.Stderr, "tag=<tag2> count=<count2>")
+	fmt.Fprintln(os.Stderr, "...")
+	fmt.Fprintln(os.Stderr)
+	fmt.Fprintf(os.Stderr, "flags:")
+	flags.flagSet.PrintDefaults()
+	fmt.Fprintln(os.Stderr)
+	fmt.Fprintln(os.Stderr, "EXAMPLES:")
+	fmt.Fprintln(os.Stderr, "  mt-kafka-mdm-report-out-of-order -group-by-name -config metrictank.ini -partition-from 0")
+	fmt.Fprintln(os.Stderr, "  mt-kafka-mdm-report-out-of-order -metric-interval 30 -reorder-window 5 -group-by-tag namespace -config metrictank.ini -partition-from 0 -partition-to 3")
+}
+
+func ParseFlags() Flags {
+	flags := NewFlags()
+
+	flag.Usage = flags.Usage
+
+	if len(os.Args) == 2 && (os.Args[1] == "-h" || os.Args[1] == "--help") {
+		flag.Usage()
+		os.Exit(0)
+	}
```

Comment:
I don't think this check is necessary. [FlagSet.Parse](https://pkg.go.dev/flag#FlagSet.Parse) should handle this already.

Reply:
This is actually an interesting case, because without this check the FlagSet.Parse will only print out the default usage for the FlagSet as opposed to the FlagSet::Usage (despite it being defined above).

Reply:
Setting the FlagSet's usage should correct this.